tools/codegen/selective_build/selector.py

```diff
@@ -0,0 +1,112 @@
+    @abstractmethod
+    def is_operator_selected(self, name: OperatorName) -> bool:
+        pass
+
+    @abstractmethod
+    def is_operator_selected_for_training(self, name: OperatorName) -> bool:
+        pass
+
+    @abstractmethod
+    def is_root_operator(self, name: OperatorName) -> bool:
+        pass
+
+
+class SelectiveBuildV1(SelectiveBuildBase):
+    def __init__(self, operators: Dict[str, SelectiveBuildOperator]):
+        self.operators = operators
+
+    def is_operator_selected(self, name: OperatorName) -> bool:
+        # For V1, check just the base operator name since the overload names are missing.
+        return str(name.name) in self.operators
+
+    def is_operator_selected_for_training(self, name: OperatorName) -> bool:
+        if not self.is_operator_selected(name):
+            return False
+        op: SelectiveBuildOperator = self.operators[name.name]
+        return op.is_used_for_training
+
+    def is_root_operator(self, name: OperatorName) -> bool:
+        if not self.is_operator_selected(name):
+            return False
+        op: SelectiveBuildOperator = self.operators[name.name]
+        return op.is_root_operator
+
+class SelectiveBuildV2(SelectiveBuildBase):
+    def __init__(self, operators: Dict[str, SelectiveBuildOperator]):
+        self.operators = operators
+
+    def is_operator_selected(self, name: OperatorName) -> bool:
+        # Check w/o stripping the overload name
+        return str(name) in self.operators
+
+    def is_operator_selected_for_training(self, name: OperatorName) -> bool:
+        if not self.is_operator_selected(name):
+            return False
+        op = self.operators[name.name]
+        return op.is_used_for_training
+
+    def is_root_operator(self, name: OperatorName) -> bool:
+        if not self.is_operator_selected(name):
+            return False
+        op: SelectiveBuildOperator = self.operators[name.name]
+        return op.is_root_operator
+
+class SelectiveBuildNotSelective(SelectiveBuildBase):
+    def __init__(self):
+        pass
+
+    def is_operator_selected(self, name: OperatorName) -> bool:
+        return True
+
+    def is_operator_selected_for_training(self, name: OperatorName) -> bool:
+        return True
+
+    def is_root_operator(self, name: OperatorName) -> bool:
+        return True
+
+class SelectiveBuildSelectorFactory(object):
+    @staticmethod
+    def create_from_path(config_path: str, version: str) -> SelectiveBuildBase:
+        with open(config_path, 'r') as f:
+            contents = yaml.load(f)
+            return SelectiveBuildSelectorFactory.create_from_contents(contents, version)
+
+    @staticmethod
+    def create_from_contents(config_contents: str, version: str) -> SelectiveBuildBase:
+        contents = yaml.load(config_contents)
+        return SelectiveBuildSelectorFactory.create_from_dict(contents, version)
+
+    @staticmethod
+    def create_from_dict(config_dict: Optional[Any], version: str) -> SelectiveBuildBase:
```

Comment:
It'd be more type safe to use a Union type here; or if this is coming from an external source (like a YAML file) declare the type to be an `object` and then add in the necessary dynamic checks to get it to be a dict. I find that when I avoid use of Any, the type checker helps me put in all of the input checking that I need, and this generally leads to better and more comprehensive error checking for malformed inputs.